### PR TITLE
Add setter for GeoJsonDataSource.name

### LINKED ISSUE #5653: Add setter for GeoJsonDataSource.name
It would be nice to be able to set the name for a DataSource, as currently, an error isn't even thrown, but the code following a call to `data.name` is completely ignored.

Consider this code

    let data_promise = Cesium.GeoJsonDataSource.load(url);
    data_promise.then(data => {
        viewer.dataSources.add(data);
        console.log(1);
        data.name = "Test";
        console.log(2);
    });

In this case, the `2` would not get printed, and the name of the dataSource wouldn't get updated. At the very least, I think an error should pop up so that people aren't left wondering why their code isn't executing, but being able to set the name for the layer would be a good thing to have, and should be very easy to implement.

### PR DESCRIPTION
**Pull request by hpinkos**
Fixes #5653

### CODE REVIEW

**Comment by hpinkos** [comment]
@mramato do you have any idea why CI is failing?  It's doing that on a couple of my branches

File: Source/DataSources/GeoJsonDataSource.js [hunk 1]
```diff
@@ -690,6 +690,12 @@ define([
         name : {
             get : function() {
                 return this._name;
+            },
+            set : function(value) {
```
Comment (burn123) [inline Source/DataSources/GeoJsonDataSource.js]: Should probably update the comment above to indicate the name can be changed

Comment (hpinkos) [inline Source/DataSources/GeoJsonDataSource.js]: Good call, thanks!  I just pushed an update to the doc

Comment (mramato) [inline Source/DataSources/GeoJsonDataSource.js]: I need to double check in the code, but I don't think we want to raise the `_changed` event, here.  I think that event is meant for underlying Entity changes, which would basically cause more work to be done than needed if we raise it here.  But like I said, I need to check how we are using it.

Comment (hpinkos) [inline Source/DataSources/GeoJsonDataSource.js]: I copied this from `CustomDataSource` FYI

Comment (mramato) [inline Source/DataSources/GeoJsonDataSource.js]: OK, looking at the code we don't seem to use this internally other than setting the DataSource clock in viewer, so this is fine as is.

**Pull request merged**